pkg/misc/depversion/depversion_test.go

```diff
@@ -242,6 +242,14 @@ func Test_VersionRangeParse(t *testing.T) {
 				},
 			},
 		},
+		{
+			input: "~0.10.x",
+			expect: VersionMatchObject{
+				VRSet: []VersionRange{
+					{">=0.10.0,<0.11.0"},
+				},
+			},
+		},
```

Comment:
Looks like NPM has some weird cases where you can do things like `^1.3.x` which technically means `>=1.0, <2.0 `which is not obvious, can we help to add these checks and make sure we handle them properly?

```
^1.0.x ==> ">=1.0, <2.0
~1.x ==> ">=1.0, <2.0
^1.0.x ==> ">=1.0, <2.0
```
```

Reply:
@lumjjb I don’t understand the `^`. How does `^1.3.x` mean `>=1.0,<2.0`?

Reply:
neither do i... it should really be invalid, but unfortunately, that's allowed in NPM so we put it in and ran it and it seemed to behave that way... so those test cases we have above are based on our testing with npm.

Reply:
So, how would to get these cases to work, should I remove the wildcard, because when I remove the `^`, I just get a regular wildcard case. For example when I run `^1.3.x` I get the output of `">=1.0.0,<1.1.0"`.

Reply:
>For example when I run ^1.3.x after removing the ^, I get the output of ">=1.0.0,<1.1.0".

Do you mean after removing the x? Removing `^` should get ">=1.3.0, <1.4.0" i think.


it looks like between the prefix and the wildcard, it takes the broader range. at least that's what i perceive the behavior to be (not documented unfortunately). 

One way to handle that i think it that if there is a prefix, then for the parsed wildcard, setting that particular version octet to x would handle this case. 

For example for ^1.3.x, major = 1, minor = 3, patch = x, in this case, since "^" is the prefix, set minor = x. 

The other way is to parse both and pick the wider range. That may be easier if there are some helpers to do so in the semver library.

Reply:
@lumjjb These are my current test cases:

```
^1.0.x -> ">=1.0,<2.0"
^1.x -> ">=1.0,<2.0"
^1.3.x -> ">=1.0,<2.0"
```

Can I change it to be (Including the patch): 

```
^1.0.x -> ">=1.0.0,<2.0.0"
^1.x -> ">=1.0.0,<2.0.0"
^1.3.x -> ">=1.0.0,<2.0.0"
```

Or does it have to stay at only the minor version?

Reply:
yea that's fine!

Reply:
Thanks, done that!